src/Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration/Extensions/XElementExtensions.cs

```diff
@@ -25,35 +25,42 @@ public static string GetDescriptionText(this XElement element)
             var description = new StringBuilder();
 
             var children = element.Nodes();
+
             foreach (var child in children)
             {
-                if (child.NodeType == XmlNodeType.Text)
-                {
-                    description.Append(child);
-                }
-                else if (child.NodeType == XmlNodeType.Element)
+                switch (child.NodeType)
                 {
-                    var childElement = (XElement)child;
-                    if (childElement.Name == KnownXmlStrings.Para)
-                    {
-                        description.Append(GetDescriptionText(childElement));
-                    }
-                    else if (childElement.Name == KnownXmlStrings.See)
-                    {
-                        description.Append(childElement.Attribute(KnownXmlStrings.Cref)?.Value);
-                    }
-                    else if (childElement.Name == KnownXmlStrings.Seealso)
-                    {
-                        description.Append(childElement.Attribute(KnownXmlStrings.Cref)?.Value);
-                    }
-                    else if (childElement.Name == KnownXmlStrings.Paramref)
-                    {
-                        description.Append(childElement.Attribute(KnownXmlStrings.Name)?.Value);
-                    }
-                    else if (childElement.Name == KnownXmlStrings.Typeparamref)
-                    {
-                        description.Append(childElement.Attribute(KnownXmlStrings.Name)?.Value);
-                    }
+                    case XmlNodeType.Text:
+                        description.Append(child);
+                        break;
+
+                    case XmlNodeType.CDATA:
+                        description.Append(child.ToString().Replace("<![CDATA[","").Replace("]]>",""));
```

Comment:
>Replace [](start = 60, length = 7)

If you're confident using Replace will always behave the way you want, I'm fine with this code. I just have reservations about it correctly working because I don't know enough about CDATA; for instance, can multiple CDATA be present / nested and this will behave the way you want?

Reply:
Tried nesting a CData tag in another CData tag and its not valid, i am getting badly format xml error. So this should work.

---
In reply to: [187713884](https://github.com/Microsoft/OpenAPI.NET.CSharpAnnotations/pull/120#discussion_r187713884) [](ancestors = 187713884)

Reply:
If there are multiple CData tags this solution works.

---
In reply to: [187715945](https://github.com/Microsoft/OpenAPI.NET.CSharpAnnotations/pull/120#discussion_r187715945) [](ancestors = 187715945,187713884)